servo/connectors/kubernetes.py

```diff
@@ -1857,8 +1873,7 @@ def _check_conditions(self, conditions: List[kubernetes_asyncio.client.V1Deploym
                 elif condition.status == "False":
                     raise servo.AdjustmentRejectedError(
                         "ProgressionFailure: message='{condition.status.message}', reason='{condition.status.reason}'",
```

Comment:
should use f-string

Reply:
Good catch

Reply:
This also means that we are missing test coverage... we should consider using mocking or something to trigger these code paths if each one of the errors is tough to reproduce as an integration test.

Maybe we factor this off of being an instance method into a standalone function or a new class where we can just unit test it?

Reply:
Fixed f-string

Opened SOL-356 for Blake's additional note